### PR TITLE
Update developer guide (link)

### PR DESCRIPTION
**Pull request by shwene**
Updated developer guide with the implementation details of the add-link feature.

The attached activity diagram for the add-link feature is a draft.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -156,6 +156,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add link feature
+
+The 'add link' feature allows for the user to add links with corresponding aliases to a `Module` in Plannit.
```
Comment (Tan-Jia-Rong) [inline docs/DeveloperGuide.md:161]: @AY2223S1-CS2103T-T10-1/developers 

Noticed that ' is used 'add link' to highlight the feature you are implementing. Would like to seek a general consensus on what we should adopt towards highlighting the features

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -156,6 +156,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+The add link mechanism is facilitated by `AddLinkCommandParser` and `AddLinkCommand`.
+`AddLinkCommandParser` implements the `Parser` interface to validate the `Module` and
+pair link URLs with their aliases from the user input.
+It then creates a `AddLinkCommand` object with the pairings for the specified `Module`.
+`AddLinkCommand` extends the `Command` class to add `Link` objects into a `Module` in Plannit.
+
+Given below is an example usage scenario and how the add link mechanism behaves at each step.
+
+**Step 1**: The user decides to add a link to a current module in Plannit using the following input:
+`add-link m/CS1231 l/<link URL> la/<link alias>`.
```
Comment (Tan-Jia-Rong) [inline docs/DeveloperGuide.md:178]: @AY2223S1-CS2103T-T10-1/developers 

Require a general consensus of whether we should include example input or how we should write the example input

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -156,6 +156,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+**Step 5**: The `AddLinkCommand::execute` method is then called by the `LogicManager`.
+This method will first obtain the `Module` object with the module code indicated by the user.
+A copy of the `Module`'s fields are then created and the `Link` object is added to the copied `TreeMap` of links.
+
+**Step 6**: A new `Module` is created with the modified and copied fields, which replaces
+the original `Module` object in Plannit using the `Model::setModule` method.
+
+The following sequence diagram shows how the 'add link' feature works:
+![AddLinkSequenceDiagram](images/AddLinkFeature/AddLinkSequenceDiagram.png)
```
Comment (Tan-Jia-Rong) [inline docs/DeveloperGuide.md:200]: Sequence diagram missing destroyer marker

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -156,6 +156,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+The following activity diagram shows how the 'add link' feature works:
+![AddLinkActivityDiagram](images/AddLinkFeature/AddLinkActivityDiagram.png)
+
+#### Design considerations:
```
Comment (Tan-Jia-Rong) [inline docs/DeveloperGuide.md:205]: Should we include rationale over current choice? @AY2223S1-CS2103T-T10-1/developers

**Review by Tan-Jia-Rong — COMMENTED** [review]
LGTM just some minor details that we should discuss as a group!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (cheeheng) [reply docs/DeveloperGuide.md:178]: From AB3 input, it looks like it is fine to truncate the command using ... to make it shorter.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -156,6 +156,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add link feature
+
+The 'add link' feature allows for the user to add links with corresponding aliases to a `Module` in Plannit.
+Links in Plannit are represented by the `Link` class. A `Link` object contains two required fields,
+a `String` URL and a `String` alias. For each Plannit `Module`, `Link` objects are stored in a `TreeMap`.
+
+The implementation of the 'delete link' feature is highly similar to this, but without a String URL input.
```
Comment (teoyuqi) [inline docs/DeveloperGuide.md:165]:
```suggestion
The implementation of the 'delete link' feature is highly similar to this, but without a `String` URL input.
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -156,6 +156,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+Thus, a new `AddLinkCommandParser` object is instantiated to parse the arguments from the user input
+to create a new `AddLinkCommand` object.
+
+**Step 4**: Within the new `AddLinkCommandParser` object, the `parse` method is used on the arguments to validate
+its module code, link URL, and alias. Also, it forms a new `Link` object with its link URL and alias.
+A new `AddLinkCommand` is created with the module code and `Link` object, which is returned to `LogicManager`.
+
+**Step 5**: The `AddLinkCommand::execute` method is then called by the `LogicManager`.
+This method will first obtain the `Module` object with the module code indicated by the user.
+A copy of the `Module`'s fields are then created and the `Link` object is added to the copied `TreeMap` of links.
```
Comment (teoyuqi) [inline docs/DeveloperGuide.md:194]:
```suggestion
A copy of the `Module`'s fields is then created and the `Link` object is added to the copied `TreeMap` of links.
```

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (teoyuqi) [reply docs/DeveloperGuide.md:205]: Yep, I think it would be clearer to the reader if we include it for our final version

**Review by teoyuqi — COMMENTED** [review]
Just some minor typos

**Review by cheeheng — COMMENTED** [review]
LGTM, but sequence diagram still missing destroyer marker (see warning message about PlantUML limitations on AB3).

**Review by cheeheng — APPROVED** [review]
LGTM

**Review by teoyuqi — APPROVED** [review]
LGTM

**Pull request merged**